src/workerd/api/web-socket.c++

```diff
@@ -400,8 +438,10 @@ kj::Promise<void> WebSocket::Accepted::createAbortTask(Native& native, IoContext
       LOG_EXCEPTION("webSocketWhenAborted", e);
     }
   }).then([this, &native]() {
-    // Other end disconnected prematurely. We may be able to clean up our state.
-    native.outgoingAborted = true;
+    if (!native.closedOutgoing) {
```

Comment:
Why not just set `error` to prevent `reportError` from dispatching another event?

Reply:
Mostly because I have no idea what I'm doing. :sweat_smile: 

Alright, I set `error` before dispatching the error and undid this hunk. It seems to work as expected, so thanks for the clue!